### PR TITLE
Website sitemap

### PR DESCRIPTION
**Pull request by bladey**
Adds `sitemap.xml` and `robots.txt` to build process, generating files dynamically in the public folder when deploying to Vercel.

image

image

### CODE REVIEW

File: docs/lib/next-sitemap.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  siteUrl: 'https://next.keystonejs.com',
+  generateRobotsTxt: true,
+};
+
+export {};
```
Comment (bladey) [inline docs/lib/next-sitemap.ts:6]: Required for ESLint, better place for this somewhere?

File: docs/package.json [hunk 2]
```diff
@@ -5,8 +5,9 @@
   "license": "MIT",
   "scripts": {
     "dev": "next -p 8000",
-    "build": "next build",
+    "build": "next build && yarn sitemap",
     "start": "next start -p 8000",
+    "sitemap": "yarn next-sitemap --config lib/next-sitemap.ts",
```
Comment (bladey) [inline docs/package.json:10]: Built after Vercel runs `next build`, confirmed the `sitemap.xml` and `robots.txt` get generated and are accessible from the preview URL.

File: docs/package.json [hunk 3]
```diff
@@ -5,8 +5,9 @@
   "license": "MIT",
   "scripts": {
     "dev": "next -p 8000",
-    "build": "next build",
+    "build": "next build && yarn sitemap",
     "start": "next start -p 8000",
+    "sitemap": "yarn next-sitemap --config sitemap.js",
```
Comment (bladey) [inline docs/package.json:10]: Generates `sitemap.xml` and `robots.txt` in the public directory after `next build` which are then accessible.

**Review by timleslie — APPROVED** [review]
👍

**Review by raveling — APPROVED** [review]

**Pull request merged**